### PR TITLE
added a request approval reset button to the action drop down of mode…

### PR DESCRIPTION
**Pull request by ARADDCC002**
…l and deployment overviews

### CODE REVIEW

File: pages/model/[uuid].tsx
Comment (a3957273) [inline pages/model/[uuid].tsx:153]: We need to work out how we're handling errors, automatic retrying, etc.  Could you raise a ticket for this?

Comment (a3957273) [reply pages/model/[uuid].tsx:153]: (Can just be a one-liner to remind us on the elaboration board)

File: pages/model/[uuid].tsx
Comment (a3957273) [inline pages/model/[uuid].tsx:234]: If this logic starts getting any more complicated we should think about moving it above into an intermediary variable.  It's fine for now, :)

File: pages/deployment/[uuid].tsx
Comment (a3957273) [reply pages/deployment/[uuid].tsx:138]: I've just noticed that `deployment` here is the full deployment object including all the metadata, etc?  I think we should actually only be posting `{ uuid: deployment.uuid }` in retrospect.

Comment (a3957273) [reply pages/deployment/[uuid].tsx:138]: Alternatively, post no data at all.  Just an empty object.  We already have the deployment uuid.

File: server/models/Deployment.ts
Comment (a3957273) [inline server/models/Deployment.ts:16]: 👍

File: server/routes/v1/deployment.ts
Comment (a3957273) [inline server/routes/v1/deployment.ts:125]: I'd like to see the deployment that the user is requesting here.  So the first parameter should be `{ deploymentUuid: body.uuid }`.

File: server/routes/v1/deployment.ts
Comment (a3957273) [inline server/routes/v1/deployment.ts:123]: You _cannot_ cast to `Deployment` here, as it's a `Deployment | null` type.  You _must_ check that Deployment is a thing, or this function will error when it's not found.

File: server/routes/v1/deployment.ts
Comment (a3957273) [reply server/routes/v1/deployment.ts:125]: Also, the right hand side of these error messages are user facing, so maybe make it more descriptive (e.g. `You cannot reset the approvals for a deployment you do not own.`).

File: server/routes/v1/deployment.ts
Comment (a3957273) [inline server/routes/v1/deployment.ts:124]: I'm worried about id changes, whenever that becomes a requirement.  Please could we compare the database id with the deployment owner (e.g. `user?._id !== deployment.owner`).

File: server/routes/v1/deployment.ts
Comment (a3957273) [inline server/routes/v1/deployment.ts:132]: Include details like deploymentId, what version was requested.

File: server/routes/v1/deployment.ts
Comment (a3957273) [inline server/routes/v1/deployment.ts:123]: `{ uuid: body.uuid }`, mentioned above it makes no sense to send entire document.  Instead, take this from `req.params.uuid`.

**Review by a3957273 — CHANGES_REQUESTED** [review]

**Comment by a3957273** [comment]
I think this should be disabled on models / deployments that are _already_ approved.  I can't think of a situation where a double approved model would want to restart the process?

File: pages/deployment/[uuid].tsx
Comment (ARADDCC002) [reply pages/deployment/[uuid].tsx:138]: I think I did this so we didn't need to make a DB call in order to get the deployment data on the server-side, as we already have the information we need in the client-side. I'll make the changes, though.

File: server/routes/v1/deployment.ts
Comment (ARADDCC002) [reply server/routes/v1/deployment.ts:123]: Resolved

File: server/routes/v1/deployment.ts
Comment (ARADDCC002) [reply server/routes/v1/deployment.ts:125]: Improved the error text.

File: server/routes/v1/deployment.ts
Comment (ARADDCC002) [reply server/routes/v1/deployment.ts:123]: Resolved

**Comment by ARADDCC002** [comment]
> I think this should be disabled on models / deployments that are _already_ approved. I can't think of a situation where a double approved model would want to restart the process?

The problem with this is that deployments only have one approval, if there's no approvals then you can't reset, and if it has been approved then in this scenario you wouldn't be able to reset it either. Unless of course it's rejected? But if it's rejected I assume you'd need to edit it that would reset the approvals anyway. 

I think there might be edge cases whereby people want to reset approvals in the case that a mistake has been spotted after it's been approved.

**Review by a3957273 — APPROVED** [review]
Looks great to me, 👍

**Pull request merged**
